packages/hash/subgraph/src/types/identifier.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  EntityEditionId,
+  EntityIdAndTimestamp,
+  OntologyTypeEditionId,
+  GraphElementEditionId,
+} from "@hashintel/hash-graph-client";
+import { validate } from "uuid";
+
+// ${AccountId}%${EntityUuid}`
+export type EntityId = `${string}%${string}`;
```

Comment:
I had created this before but I figured it might be confusing when we actually do have an `AccountId` type in the HASH workspace and such, and this functionally doesn't change anything because they are just plain strings.

I looked into a string template literal type for UUIDs [but it's insane](https://stackoverflow.com/questions/68724603/how-to-create-a-uuid-template-literal-type-in-typescript)

Reply:
> I figured it might be confusing when we actually do have an AccountId type in the HASH workspace and such

We do?  I'm not aware of an `AccountId` in the workspace.

> this functionally doesn't change anything because they are just plain strings

You're right that the functionality doesn't change, my only point is that in my view `${AccountId}%${EntityUuid}` is easier to read/understand than `${string}%${string}`.

> I looked into a string template literal type for UUIDs [but it's insane](https://stackoverflow.com/questions/68724603/how-to-create-a-uuid-template-literal-type-in-typescript)

Agreed.

Reply:
> We do? I'm not aware of an AccountId in the workspace.

I thought we did but it would seem we just use `string` all over the place frustratingly. That would make me even less inclined to add it here, because we would either keep it private, which would make it strange that we aren't then using it across the workspace (and potentially someone would unknowingly make it public), or we'd export it, and then we'd have to consider why we aren't using it all across the workspace in places like `ownedById: string`, etc.

I'd prefer if we leave it as `string` for now and address this later on if we want to encapsulate `AccountId` as a type (which I think we should probably do, but it'd be a big change)

Reply:
> I'd prefer if we leave it as string for now and address this later on if we want to encapsulate AccountId as a type (which I think we should probably do, but it'd be a big change)

Understood, maybe it's worth a `@todo` here then?